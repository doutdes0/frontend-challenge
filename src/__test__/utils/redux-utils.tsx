import { setUpStore } from "../../redux/store";
import { Provider } from "react-redux";
import { AppStore, RootState } from "../../redux/store";
import { render, RenderOptions } from "@testing-library/react";

interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
  preloadedState?: Partial<RootState>;
  store?: AppStore;
}

export const renderWithProvider = (
  component: React.ReactElement,
  {
    preloadedState = {},
    store = setUpStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) => {
  const _provider = ({ children }: { children: React.ReactNode }) => {
    return <Provider store={store}>{children}</Provider>;
  };
  return {
    store,
    ...render(component, { wrapper: _provider, ...renderOptions }),
  };
};

export const key = "67b3f85d-5354-4bd6-ba9a-5482fb966057";
export const preloadedState: RootState = {
  cats: {
    "67b3f85d-5354-4bd6-ba9a-5482fb966057": {
      id: "67b3f85d-5354-4bd6-ba9a-5482fb966057",
      url: "https://cdn2.thecatapi.com/images/2n3.jpg",
      isLiked: false,
    },
  },
};
