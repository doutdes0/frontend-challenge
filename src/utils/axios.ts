import axios from "axios";

export const catnipThrower = axios.create({
  baseURL: " https://api.thecatapi.com/v1/images/search",
  params: {
    limit: 15,
    api_key: `${process.env.VITE_API_KEY}`,
  },
});
