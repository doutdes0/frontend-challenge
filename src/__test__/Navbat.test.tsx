import NavBar from "../components/Navbar";
import { MemoryRouter, BrowserRouter } from "react-router-dom";
import { renderWithProvider, preloadedState } from "./utils/redux-utils";
import { fireEvent } from "@testing-library/react";

describe("Navbar", () => {
  it("should match snapshot", () => {
    const { container } = renderWithProvider(
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>,
      { preloadedState }
    );
    expect(container).toMatchSnapshot();
  });
  describe("Link", () => {
    it("should be active onload on home route, should toggle active class on route change", () => {
      const { container } = renderWithProvider(
        <MemoryRouter initialEntries={["/"]}>
          <NavBar />
        </MemoryRouter>,
        { preloadedState }
      );

      const homeLink = container.querySelector('div[data-link="/"] a');
      const favLink = container.querySelector('div[data-link="/favorites"] a');

      expect(homeLink?.classList.contains("active")).toBeTruthy();
      fireEvent.click(favLink!);
      expect(homeLink?.classList.contains("active")).not.toBeTruthy();
      expect(favLink?.classList.contains("active")).toBeTruthy();
    });
  });
});
