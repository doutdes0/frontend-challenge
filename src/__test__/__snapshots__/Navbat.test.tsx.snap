// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should match snapshot 1`] = `
<div>
  <div
    class="sc-beySPh kqHCnd"
  >
    <div
      class="link active"
      data-link="/"
    >
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        Все котики
      </a>
    </div>
    <div
      class="link"
      data-link="/favorites"
    >
      <a
        class=""
        href="/favorites"
      >
        Любимые котики
      </a>
    </div>
  </div>
</div>
`;
